File: avalon-modules/avalon-dataset/avalon-dataset-repository/src/main/java/io/github/lishangbu/avalon/dataset/repository/GrowthRateRepository.java
package io.github.lishangbu.avalon.dataset.repository;

import io.github.lishangbu.avalon.dataset.entity.GrowthRate;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
import org.springframework.stereotype.Repository;

/// 成长速率(GrowthRate)数据访问层
///
/// 提供基础的 CRUD 操作
///
/// @author lishangbu
/// @since 2026/2/10
@Repository
public interface GrowthRateRepository
        extends JpaRepository<GrowthRate, Long>, JpaSpecificationExecutor<GrowthRate> {}

File: avalon-modules/avalon-dataset/avalon-dataset-repository/src/main/kotlin/io/github/lishangbu/avalon/dataset/repository/GrowthRateRepository.kt
package io.github.lishangbu.avalon.dataset.repository

import io.github.lishangbu.avalon.dataset.entity.GrowthRate
import org.babyfish.jimmer.Specification
import org.babyfish.jimmer.spring.repository.KRepository

/**
 * 成长速率仓储接口
 *
 * 定义成长速率数据的查询与持久化操作
 *
 * @author lishangbu
 * @since 2026/2/10
 */
interface GrowthRateRepository : KRepository<GrowthRate, Long> {
    /** 按条件查询成长速率列表 */
    fun findAll(specification: Specification<GrowthRate>?): List<GrowthRate> =
        sql
            .createQuery(GrowthRate::class) {
                specification?.let(::where)
                select(table)
            }.execute()
}
